crypto/x509/x509_obj.c

```diff
@@ -110,11 +111,16 @@ char *X509_NAME_oneline(const X509_NAME *a, char *buf, int len)
                 continue;
             l2++;
 #ifndef CHARSET_EBCDIC
-            if ((q[j] < ' ') || (q[j] > '~'))
+            n = q[j];
+            if (n == '/' || n == '+')
+                l2++; /* char needs to be escaped */
+            else if ((n < ' ') || (n > '~'))
                 l2 += 3;
 #else
-            if ((os_toascii[q[j]] < os_toascii[' ']) ||
-                (os_toascii[q[j]] > os_toascii['~']))
+            n = os_toascii[q[j]];
+            if (n == os_toascii['/'] || n == os_toascii['+'])
+                l2++; /* char needs to be escaped */
+            else if ((n < os_toascii[' ']) || (n > os_toascii['~']))
                 l2 += 3;
 #endif
```

Comment:
Why we still have this conditional is beyond me.  A study of `include/crypto/ctype.h` shows that it's completely unnecessary.  Please apply this diff instead:

``` diff
diff --git a/crypto/x509/x509_obj.c b/crypto/x509/x509_obj.c
index c1e893bf13..86187a86bf 100644
--- a/crypto/x509/x509_obj.c
+++ b/crypto/x509/x509_obj.c
@@ -109,14 +109,11 @@ char *X509_NAME_oneline(const X509_NAME *a, char *buf, int len)
             if (!gs_doit[j & 3])
                 continue;
             l2++;
-#ifndef CHARSET_EBCDIC
-            if ((q[j] < ' ') || (q[j] > '~'))
-                l2 += 3;
-#else
-            if ((os_toascii[q[j]] < os_toascii[' ']) ||
-                (os_toascii[q[j]] > os_toascii['~']))
+            if (q[j] == '/' || q[j] == '+')
+                l2++; /* char needs to be escaped */
+            else if ((ossl_toascii(q[j]) < ossl_toascii(' ')) ||
+                     (ossl_toascii(q[j]) > ossl_toascii('~')))
                 l2 += 3;
-#endif
         }
 
         lold = l;
```

Reply:
You need to add aan `#include "crypto/ctype.h"` as well, of course

Reply:
Done - yet I had to use `_openssl_os_toascii` instead of `ossl_toascii`
and had to declare before:
```
#ifndef CHARSET_EBCDIC
# define _openssl_os_toascii(c) c
#endif
```

Reply:
> You need to add aan `#include "crypto/ctype.h"` as well, of course

Ah, I had not seen this further comment at first. Using it, your patch works as given.